xonsh/environ.py

```diff
@@ -106,7 +106,6 @@ def to_debug(x):
     re.compile('\w*PATH$'): (is_env_path, str_to_env_path, env_path_to_str),
     'PATHEXT': (is_env_path, str_to_env_path, env_path_to_str),
     'RAISE_SUBPROC_ERROR': (is_bool, to_bool, bool_to_str),
-    'RIGHT_PROMPT': (is_string, ensure_string, ensure_string),
```

Comment:
So, I generally think it is a bad idea to remove the validators.  That said, these validators are clearly wrong.  What we need are `string_or_callable()` validators and associated tests.  Is this something that you would be willing to implement.  The other validation functions live in `xonsh.tools`.  If you add these:

1. I'd *really* appreciate it, and
2. We'd be able to use them for $PROMPT and `$MULTILINE_PROMPT` too!